packages/nx/src/native/native-file-cache-location.ts

```diff
@@ -1,10 +1,24 @@
-import { tmpdir } from 'os';
+import { tmpdir, userInfo } from 'os';
 import { join } from 'path';
 import { createHash } from 'crypto';
 import { workspaceRoot } from '../utils/workspace-root';
 
 export const nativeFileCacheLocation = join(
-  tmpdir(),
-  'nx-native-file-cache',
-  createHash('sha256').update(workspaceRoot).digest('hex')
+  getNativeFileCacheBaseFolder(),
+  createHash('sha256')
+    .update(workspaceRoot)
+    .update(userInfo().username)
+    .digest('hex')
 );
+
+function getNativeFileCacheBaseFolder() {
+  if (process.env.NX_NATIVE_FILE_CACHE_DIRECTORY) {
+    return process.env.NX_NATIVE_FILE_CACHE_DIRECTORY;
+  } else {
+    const shortUserHash = createHash('sha256')
+      .update(userInfo().username)
+      .digest('hex')
+      .substring(0, 7);
+    return join(tmpdir(), `nx-native-file-cache-${shortUserHash}`);
```

Comment:
Do we need a `shortUserHash` here as well as in the `hash` for the child directory?

right now we have...

`/tmp/[F(username)]/[F(username, workspaceRoot]`

We should be able to have..

`/tmp/[F(username, workspaceRoot)]` right?